### ISSUE TITLE
Memory leak in wolfssl client on HelloRetryRequest

### ISSUE THREAD

**Issue by tatsuhiro-t** [ngtcp2/ngtcp2#510]
When wolfssl client receives HelloRetryRequest from server, it leaks memory:

```
==32398==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 16 byte(s) in 1 object(s) allocated from:
    #0 0x55f2ceb060ae in malloc (/path/ngtcp2/examples/wsslclient+0xd50ae)
    #1 0x7ff5aa3e0fc7 in QuicTransportParam_dup (/path/lib/libwolfssl.so.34)

Indirect leak of 70 byte(s) in 1 object(s) allocated from:
    #0 0x55f2ceb060ae in malloc (/path/ngtcp2/examples/wsslclient+0xd50ae)
    #1 0x7ff5aa3e0fd8 in QuicTransportParam_dup (/path/lib/libwolfssl.so.34)

SUMMARY: AddressSanitizer: 86 byte(s) leaked in 2 allocation(s).
```

How to repro:

1. Run server: examples/server with --groups X25519
2. Run wssl client: examples/wsslclient

@icing

**Comment by icing** [comment]
Reproduced this and made a PR with fix at https://github.com/wolfSSL/wolfssl/pull/5471

**Comment by icing** [comment]
Fix was merged in 3f7d56d957ac7418abab13540af5e4a6c8ba364a of wolfSSL.

**Comment by tatsuhiro-t** [comment]
I confirmed that the upstream has fixed this issue.  Thank you very much.
